CMakeLists.txt

```diff
@@ -14,6 +14,10 @@ read_version("${CMAKE_CURRENT_SOURCE_DIR}/version.pri" VER_MAJOR VER_MINOR VER_B
 
 project(qBittorrent VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD})
 
+# check for invalid compiler version/CXX standard as early as possible
+include(FunctionQbtCXXCompilerAndModeCheck)
```

Comment:
Does it matter for the caller is it a function or a macro? If no, maybe just include(QbtCXXCompilerAndModeCheck)?

Reply:
@zeule 
I would prefer it like this, since other files use the same convention, e. g. `FunctionReadVersion.cmake`, `MacroConfigureMSVCRuntime.cmake`, etc.

Reply:
What's the point to have one function/macro per file?

Reply:
Cleaner separation? I don't know. It was already done this way.

Reply:
>I don't know. It was already done this way.

Well, it's separate task to cleanup it.